app/src/main/java/com/zulip/android/filters/NarrowFilterStar.java

```diff
@@ -0,0 +1,86 @@
+package com.zulip.android.filters;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import com.j256.ormlite.stmt.SelectArg;
+import com.j256.ormlite.stmt.Where;
+import com.zulip.android.R;
+import com.zulip.android.ZulipApp;
+import com.zulip.android.models.Message;
+import com.zulip.android.models.Stream;
+
+import org.json.JSONException;
+
+import java.sql.SQLException;
+
+public class NarrowFilterStar implements NarrowFilter {
+
+    public static final Parcelable.Creator<NarrowFilterStar> CREATOR = new Parcelable.Creator<NarrowFilterStar>() {
+        public NarrowFilterStar createFromParcel(Parcel in) {
+            return new NarrowFilterStar();
+        }
+
+        public NarrowFilterStar[] newArray(int size) {
+            return new NarrowFilterStar[size];
+        }
+    };
+
+    private boolean isStarred;
+
+    public NarrowFilterStar() {
+    }
+
+    public int describeContents() {
+        return 0;
+    }
+
+    public void writeToParcel(Parcel dest, int flags) {
+    }
+
+    @Override
+    public Where<Message, Object> modWhere(Where<Message, Object> where)
+            throws SQLException {
+        where.like(Message.MESSAGE_STAR_FIELD, new SelectArg(isStarred));
+        return where;
+    }
+
+    @Override
+    public String toString() {
+        try {
+            return getJsonFilter();
+        } catch (JSONException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public boolean matches(Message msg) {
+        return msg.getFlags() != null && msg.getFlags().contains("starred");
```

Comment:
It would be better to check 
return msg.getFlags() != null && (msg.getMessageStar() || msg.getFlags().contains("starred"));